addons/sale_margin/models/sale_order.py

```diff
@@ -10,7 +10,7 @@ class SaleOrderLine(models.Model):
         "Margin", compute='_compute_margin',
         digits='Product Price', store=True, groups="base.group_user")
     margin_percent = fields.Float(
-        "Margin (%)", compute='_compute_margin', store=True, groups="base.group_user")
+        "Margin (%)", compute='_compute_margin', store=True, groups="base.group_user", group_operator="avg")
```

Comment:
the avg operator is not good.
you should by pass the readgroup function.

exemple:
line 1 : cost 10, price 20 : marge 50%
line 2: cost 10, price 100 : marge 90%

avg(50, 90) --> 70%
(20 + 100 - 10 - 10)/(20 + 100) --> 83%

The good value is 83%

@tde-banana-odoo

Reply:
the point is not to have an exact computation in all cases, the point is to have at least one computation that makes sense
e.g. you can say "on average, for customer X my margin is X%" (when grouping by partner, for ex.)
summing will *never* be correct though

overriding the read_group seems like overkill for this (imho)

although perhaps @dala should have the last word on this

Reply:
it's @dalaOdoo I think :eyes:

Reply:
The margin % is not available as a measure in the sale reporting and in the pivot view so there is no reason to allow it in the graph view at this stage. The easiest fix is to remove it from the graph view.

Reply:
@bouvyd are you around again?!

This is the proper way to compute such margin, that we are doing on invoices:

https://github.com/OCA/margin-analysis/blob/13.0/account_invoice_margin/report/account_invoice_report.py

Reply:
He's back as Product Owner now (not a dev anymore) :rocket: